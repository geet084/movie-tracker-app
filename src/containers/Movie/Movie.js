import React from 'react'
import { mapStateToProps, mapDispatchToProps } from '../Login/Login';
import { toggleFavorite } from '../../actions';
import { connect } from 'react-redux';

export const Movie = ({ poster_path }) => {
  const handleClick = (id) => {
    props.toggleFavorite(id);
  }

  return (
    <div className='movie-card' id={}>
      <button onClick={(event) => handleClick()}></button>
      <img className='movie-image' src={`http://image.tmdb.org/t/p/w342/${poster_path}`} alt="a" />
    </div>
  )
}

export const mapStateToProps = (state) => ({
  favorites: state.user.favorites
});

export const mapDispatchToProps = (dispatch) => ({
  toggleFavorite: (id) => dispatch(toggleFavorite(id)),
});

export default connect(null, mapDispatchToProps)(Movie);

// when user logs in, grab all favorites from backend using their user id and store them in redux state
// on movie cards, add active class to favorited movies --> if favorites.includes(movie.id), add active class
// when user clicks favorite icon, toggle favorite in redux state and do api call to add/remove from backend