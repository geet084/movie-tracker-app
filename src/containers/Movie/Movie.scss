.movie-card {
  margin: 0 auto;
  height: 450px;
  width: 300px;
  position: relative;
  top: -32px;
  
  &:hover {
    box-shadow: 0px 0px 15px 10px rgba(255,255,255,0.8);
  }
}

.movie-image {
    position: relative;
    top: -32px;
  width: 300px;
}

.placeholder {
  margin-top: 32px;
}

.favorite-icon {
  border: none;
  background: url('./images/star-outline.png');
  background-size: cover;
  position: relative;
  top: 40px;
  right: -260px;
  width: 30px;
  height: 30px;

  &:focus {
    outline: none;
  }

  &:hover {
    background: url('./images/star-yellow.png');
    background-size: cover;
  }
}

.favorite {
  background: url('./images/star-yellow.png');
  background-size: cover;
}