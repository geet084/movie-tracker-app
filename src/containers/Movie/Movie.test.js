import React from 'react'
import { shallow } from 'enzyme'
import Movie from './Movie'

describe('Movie', () => {
  it.skip('should', () => {
  })
})