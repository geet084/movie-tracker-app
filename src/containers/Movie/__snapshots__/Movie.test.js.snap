// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie defaults should match the snapshot with all data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movie
    movie_id={0}
    overview="Text about movie..."
    poster_path="url/2"
    release_date="02-03-2019"
    title="Aquaman"
    user={
      Object {
        "email": "a@a",
        "favorites": Array [],
        "id": 1,
        "name": "Tommy",
        "password": "password",
      }
    }
    vote_average={6.9}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="favorite-icon"
          onClick={[Function]}
        />,
        false,
        <Link
          replace={false}
          to="/movies/0"
        >
          <img
            alt="a"
            className="movie-image"
            src="http://image.tmdb.org/t/p/w342/url/2"
          />
        </Link>,
      ],
      "className": "movie-card",
      "id": 0,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "favorite-icon",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <img
            alt="a"
            className="movie-image"
            src="http://image.tmdb.org/t/p/w342/url/2"
          />,
          "replace": false,
          "to": "/movies/0",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "a",
            "className": "movie-image",
            "src": "http://image.tmdb.org/t/p/w342/url/2",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="favorite-icon"
            onClick={[Function]}
          />,
          false,
          <Link
            replace={false}
            to="/movies/0"
          >
            <img
              alt="a"
              className="movie-image"
              src="http://image.tmdb.org/t/p/w342/url/2"
            />
          </Link>,
        ],
        "className": "movie-card",
        "id": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "favorite-icon",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              alt="a"
              className="movie-image"
              src="http://image.tmdb.org/t/p/w342/url/2"
            />,
            "replace": false,
            "to": "/movies/0",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "a",
              "className": "movie-image",
              "src": "http://image.tmdb.org/t/p/w342/url/2",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
